### PR TITLE
Use tasty-bench by default

### LINKED ISSUE #61: Replace gauge with tasty-bench

### PR DESCRIPTION
**Pull request by adithyaov**
Closes #61

### CODE REVIEW

File: unicode-transforms.cabal [hunk 1]
```diff
@@ -172,11 +177,15 @@ benchmark bench
...
     , filepath >=1.0 && <2
-    , gauge >=0.2.0 && <0.3
     , path >=0.0.0 && <0.9
     , path-io >=0.1.0 && <1.7
     , text >=1.1.1 && <1.3
     , unicode-transforms
+  if flag(use-gauge)
+    build-depends: gauge >=0.2.0 && <0.3
+  else
+    build-depends: tasty-bench>= 0.2.5 && <0.3
```
Comment (Bodigrim) [inline unicode-transforms.cabal:187]:
```suggestion
    build-depends: tasty-bench>= 0.2.5 && <0.4
```
`tasty-bench-0.3` is already out

**Comment by harendra-kumar** [comment]
Need to fix the CIs, why is `process` failing to compile on older ghc versions?

**Comment by adithyaov** [comment]
`process` is consistently failing. We need to fix it asap if possible or defer this PR till we can put in the time to find what the problem is.

**Comment by Bodigrim** [comment]
`process-1.6.13.0` fails to build, https://matrix.hackage.haskell.org/#/package/process

As a tactical solution, you can either constrain `process /= 1.6.13.0`, or set `optparse-applicative -process` in `cabal.project`.

**Comment by adithyaov** [comment]
I think there is a fix incoming soon in `process`. We can wait for that.

**Comment by adithyaov** [comment]
A new release for `process` should be incoming soon.
See https://github.com/haskell/process/issues/211

**Comment by harendra-kumar** [comment]
> As a tactical solution, you can either constrain process /= 1.6.13.0, or set optparse-applicative -process in cabal.project.

Let's do this and not wait for a new release of `process`. Anyway 1.6.13.0 should never be chosen as it does not build.

File: unicode-transforms.cabal [hunk 2]
```diff
@@ -172,11 +177,20 @@ benchmark bench
...
     , unicode-transforms
+  if flag(use-gauge)
+    build-depends: gauge >=0.2.0 && <0.3
+  else
+    build-depends: tasty-bench>= 0.2.5 && <0.4
+
+                   -- unicode-transforms does not depend on process itself but
+                   -- some of its dependincies do. process-1.6.13.0 fails to
+                   -- build and this is basically a hack for the time being.
+                 , process < 1.6.13.0 && > 1.6.13.0
```
Comment (Bodigrim) [inline unicode-transforms.cabal:192]: Did you try `cabal update`? `process-1.6.13.0` is already deprecated and should not be chosen by `cabal` any longer.

Comment (adithyaov) [reply unicode-transforms.cabal]: Force pushed reverting the change. The CI script does do a `cabal update` I believe.

**Review by Bodigrim — APPROVED** [review]

**Pull request merged**